processing/src/main/java/org/apache/druid/segment/StorageAdapter.java

```diff
@@ -129,6 +141,18 @@ default RowSignature getRowSignature()
   @Nullable
   Metadata getMetadata();
 
+  /**
+   * Returns column names that this adapter's data is sorted by. Cursors returned by this adapter return rows in
+   * this ordering, using the natural comparator for the type of the column as returned by
+   * {@link #getColumnCapabilities(String)}. Includes {@link ColumnHolder#TIME_COLUMN_NAME} if appropriate.
+   */
+  List<String> getSortOrder();
+
+  default boolean isTimeOrdered()
+  {
+    return ColumnHolder.TIME_COLUMN_NAME.equals(Iterables.getFirst(getSortOrder(), null));
+  }
```

Comment:
This sounds like a good idea, although in the interests of minimizing conflicts, it'd probably be good to do this after #16533 is merged (since it moves `OrderBy` around).